libcnb-cargo/src/cross_compile.rs

```diff
@@ -1,6 +1,21 @@
 use std::ffi::OsString;
 use which::which;
 
+const MUSL_TARGET: &str = "x86_64-unknown-linux-musl";
+const MAC_BINARIES: CompilerBinaries = CompilerBinaries {
+    ld: Binary("x86_64-linux-musl-ld"),
+    cc: Binary("x86_64-linux-musl-gcc"),
+};
+const LINUX_BINARIES: CompilerBinaries = CompilerBinaries {
+    ld: Binary("musl-ld"),
+    cc: Binary("musl-gcc"),
+};
```

Comment:
We don't need the `*-ld` variants (whilst they are present in the cargo-make examples and #199, they are not needed and differ from the more commonly used configuration):
https://github.com/Malax/libcnb.rs/pull/199#discussion_r757620799